storage.go

```diff
@@ -2,35 +2,316 @@ package azfile
 
 import (
 	"context"
+	"encoding/base64"
 	"io"
+	"strconv"
 
+	"github.com/Azure/azure-storage-file-go/azfile"
+
+	"github.com/beyondstorage/go-storage/v4/pkg/iowrap"
+	"github.com/beyondstorage/go-storage/v4/services"
 	. "github.com/beyondstorage/go-storage/v4/types"
 )
 
 func (s *Storage) create(path string, opt pairStorageCreate) (o *Object) {
-	panic("not implemented")
+	rp := s.getAbsPath(path)
+
+	if opt.HasObjectMode && opt.ObjectMode.IsDir() {
+		o = s.newObject(true)
+		o.Mode |= ModeDir
+	} else {
+		o = s.newObject(false)
+		o.Mode |= ModeRead
+	}
+
+	o.ID = rp
+	o.Path = path
+
+	return o
+}
+
+func (s *Storage) createDir(ctx context.Context, path string, opt pairStorageCreateDir) (o *Object, err error) {
+	rp := s.getAbsPath(path)
+
+	attribute := azfile.FileAttributeNone
+
+	properties := azfile.SMBProperties{
+		FileAttributes: &attribute,
+	}
+
+	_, err = s.client.NewDirectoryURL(path).GetProperties(ctx)
+	if err == nil {
+		// The directory exist, we should delete it.
+		// ref: https://github.com/beyondstorage/specs/blob/master/rfcs/134-write-behavior-consistency.md
+		_, err = s.client.NewDirectoryURL(path).Delete(ctx)
```

Comment:
We don't need to delete the existing dir and recreate here. We can set the metadata for the existing dir object.